rfd/0037-desktop-access-protocol.md

```diff
@@ -0,0 +1,191 @@
+---
+authors: Andrew Lytvynov (andrew@goteleport.com)
+state: draft
+---
+
+# RFD 37 - Desktop Access - Desktop protocol
+
+## What
+
+RFD 33 defines the high-level goals and architecture for Teleport Desktop
+Access.
+
+This RFD specifies the Desktop protocol - wire protocol used between the
+OS-specific Teleport desktop service (like `windows_desktop_service`) and the
+web client.  OS-specific Teleport desktop services are responsible for
+translating native protocols or APIs (like RDP or X11) into the protocol
+described here.
+
+## Details
+
+### Goals
+
+This custom protocol is created with several goals in mind:
+- performance - the messages should be compact and fast to encode/decode
+- portability - should map easily to standard protocols like RDP
+- decoding simplicity - the parsing code should be simple enough for auditing,
+  especially when written in dynamic languages like JavaScript
+- extendability - new capabilities can be added in the future
+
+### Overview
+
+The protocol consists of if discrete messages, sent between a client and a
+server. These messages are passed over a secure, authenticated and reliable
+transport layer, like TLS or a websocket. The protocol leaves all security
+concerns (authentication, integrity, etc) to the transport layer.
```

Comment:
We need to test for cross-origin WebSocket hijacking (see https://portswigger.net/web-security/websockets/cross-site-websocket-hijacking). Since the transport channel is fully responsible for authentication, we need to ensure that different origins in the user agent cannot tamper with the websocket channel

Reply:
Thanks, added to https://github.com/gravitational/teleport/issues/7761